File: frontend/src/types/product.ts
export interface Product {
  id: string;
  category_id: string;
  name: string;
  slug: string;
  description?: string;
  unit: string;
  price: string;
  stock_quantity: string;
  image_url?: string;
  is_available: boolean;
  created_at: string;
  updated_at: string;
}

File: backend/internal/dto/product.go
package dto

type ProductInput struct {
	CategoryID    string  `json:"category_id"`
	Name          string  `json:"name"`
	Slug          string  `json:"slug"`
	Description   *string `json:"description"`
	Unit          string  `json:"unit"`
	Price         string  `json:"price"`
	StockQuantity string  `json:"stock_quantity"`
	ImageURL      *string `json:"image_url"`
	IsAvailable   *bool   `json:"is_available"`
}
